Add TFLite + Transformer example

calren opened the pull request:
Demonstrates applying an Effect generated by [TFLite's Artistic Style Transfer](https://www.tensorflow.org/lite/examples/style_transfer/overview) on a video using Transformer API.

https://github.com/user-attachments/assets/4c35b0de-d1e3-4e01-a461-6b355de51474

calren commented:
cc @droid-girl @lethargicpanda

droid-girl commented:
I checked-out the project. One minor comment: should it be possible to repeat the export with another style? At the moment the selector is available but the export button is disabled because the export completed already.
A proposal:
1) enable export button again when the first one completed 
2) if you want to keep it disabled, maybe also disable the style selector

droid-girl commented on samples/media/video/src/main/java/com/example/platform/media/video/StyleTransferShaderProgram.java:
```diff
@@ -0,0 +1,229 @@
+            float[] identityMatrix = GlUtil.create4x4IdentityMatrix();
+            glProgram.setFloatsUniform("uTexTransformationMatrix", identityMatrix);
+            glProgram.setFloatsUniform("uTransformationMatrix", identityMatrix);
+            // glProgram.setFloatsUniform("uRgbMatrix", identityMatrix);
```
remove commented out code?

droid-girl commented on samples/media/video/src/main/java/com/example/platform/media/video/StyleTransferShaderProgram.java:
```diff
@@ -0,0 +1,229 @@
+            bitmap = Bitmap.createBitmap(width, height, Bitmap.Config.ARGB_8888);
+            bitmap.copyPixelsFromBuffer(pixelBuffer);
+
+            Log.w("DEBUG", "Process frame at " + (presentationTimeUs / 1000) + " ms");
```
Should we follow the same convention as in other classes and define a TAG const and use it instead of "Debug"?

calren replied:
that makes sense, will do

droid-girl commented on samples/media/video/src/main/java/com/example/platform/media/video/StyleTransferShaderProgram.java:
```diff
@@ -0,0 +1,229 @@
+import javax.microedition.khronos.opengles.GL10;
+
+@UnstableApi
+final class StyleTransferShaderProgram extends BaseGlShaderProgram {
```
Should we aim to convert it to Kotlin? Do you aim to keep this repo Kotlin only or any guidelines on this?

calren replied:
yes, we should convert this to Kotlin even though we don't have any strict guidelines on keeping this repo kotlin only.

so i do this in a separate PR to keep it easier to review or just do it as part of this PR?

droid-girl replied:
I am ok with having a follow up PR with a Kotlin version as long as the repository owner is fine with it.

calren commented:
addressed comments, please take another look!

>  should it be possible to repeat the export with another style?
I disabled the radio buttons when the export is started. I tried exporting with another style after an initial export, but the result looked glitchy, and I'm not sure why. It seems like only the first export gives a good result, and subsequent ones always looks glitchy.

calren commented:
trying to make builds pass again on this repo before merging this (https://github.com/android/platform-samples/pull/185)

calren commented:
@SigmanZero can you give this an LGTM? It still needs an approval from someone with write access on this repo

calren commented:
@ksemenova @lethargicpanda can you take a quick look at this PR as well?

checking out `gradle/libs.versions.toml` will probably be the most useful, to make sure we're using recommended dependencies

calren commented:
merging this for now but @ksemenova @lethargicpanda let me know if there are concerns on the TFLite front and I can make follow up PRs to address
